wandb/sdk/wandb_artifacts.py

```diff
@@ -1805,15 +1790,185 @@ def _content_addressed_path(md5: str) -> FilePathStr:
+                        )
+                        break
+                else:  # didn't break
+                    raise ValueError(
+                        f"Couldn't find blob version for {manifest_entry.ref} matching "
+                        f"etag {manifest_entry.digest}."
+                    )
+        else:
+            downloader = blob_client.download_blob(version_id=version_id)
+        with cache_open(mode="wb") as f:
+            downloader.readinto(f)
+        return path
+
+    def store_path(
+        self,
+        artifact: ArtifactInterface,
+        path: Union[URIStr, FilePathStr],
+        name: Optional[str] = None,
+        checksum: bool = True,
+        max_objects: Optional[int] = None,
+    ) -> Sequence["ArtifactManifestEntry"]:
+        account_url, container_name, blob_name, query = self._parse_uri(path)
+        path = URIStr(f"{account_url}/{container_name}/{blob_name}")
+
+        if not checksum:
+            return [
+                ArtifactManifestEntry(
+                    path=LogicalFilePathStr(name or blob_name), digest=path, ref=path
+                )
+            ]
+
+        blob_service_client = self._get_module("azure.storage.blob").BlobServiceClient(
+            account_url,
+            credential=self._get_module("azure.identity").DefaultAzureCredential(),
+        )
+        blob_client = blob_service_client.get_blob_client(
+            container=container_name, blob=blob_name
+        )
+        if blob_client.exists(version_id=query.get("versionId")):
+            blob_properties = blob_client.get_blob_properties(
+                version_id=query.get("versionId")
+            )
+            return [
+                self._create_entry(
+                    blob_properties,
+                    path=LogicalFilePathStr(
+                        name or pathlib.PurePosixPath(blob_name).name
+                    ),
+                    ref=URIStr(
+                        f"{account_url}/{container_name}/{blob_properties.name}"
+                    ),
+                )
+            ]
+        else:
+            entries = []
+            container_client = blob_service_client.get_container_client(container_name)
+            max_objects = max_objects or DEFAULT_MAX_OBJECTS
+            for i, blob_properties in enumerate(
+                container_client.list_blobs(name_starts_with=f"{blob_name}/")
+            ):
+                if i >= max_objects:
+                    raise ValueError(
+                        f"Exceeded {max_objects} objects tracked, pass max_objects to "
+                        f"add_reference"
+                    )
+                suffix = pathlib.PurePosixPath(blob_properties.name).relative_to(
+                    blob_name
+                )
+                entries.append(
+                    self._create_entry(
+                        blob_properties,
+                        path=LogicalFilePathStr(str(name / suffix if name else suffix)),
```

Comment:
`name` will usually be a string here rather than a `Path` of some kind, so I think `name / suffix` will often raise an exception.

Reply:
`suffix` is always a `Path`, and `Path` implements `__rtruediv__`. See docs [here](https://docs.python.org/3/reference/datamodel.html#emulating-numeric-types).